src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -74,11 +66,10 @@ public async Task<MsalTokenResponse> AcquireTokenUsingBrokerAsync(Dictionary<str
                     throw new MsalException(MsalError.AndroidBrokerOperationFailed, ex.Message, ex);
             }
 
-            await _readyForResponse.WaitAsync().ConfigureAwait(false);
```

Comment:
Could you test what happens now if the broker is not installed but users configures WithBroker(true) ?